### DISCUSSION TITLE
How would I URL encode my highlights' text?

### DISCUSSION THREAD

**Discussion by daraul** [out-of-cheese-error/gooseberry#100] [Q&A] [answered]
I transfer my annotations into org-mode with the template below. How would I go about URL encoding `this` in the `ROAM_REFS` property? Think of javascript's `encodeURIComponent`.

```toml
annotation_template = '''
{{#each highlight}}** {{this}} {{#each ../tags}}:{{this}}{{#if @last}}:{{/if}}{{/each}}
:PROPERTIES:
:ANNOTATION_ID: {{../id}}
:ROAM_REFS:     [[{{../uri}}#:~:text={{this}}]]
:CREATED:       {{date_format "%c" ../created}}
:END:

{{../text}}
{{/each}}
'''
```

**Comment by Ninjani** [comment, marked as answer]
I added a `url_encode` helper function in PR #102, once that's merged you should be able to use `{{url_encode ../uri}}`
